src/main/java/gift/wishlist/WishlistService.java

```diff
@@ -22,31 +26,47 @@ public WishlistService(
     }
 
     public List<Product> getAllWishlists(MemberTokenDTO memberTokenDTO) {
-        return wishlistRepository.findAllByMemberEmail(memberTokenDTO.getEmail())
+        return wishlistRepository
+            .findAllByMember(Member.fromMemberTokenDTOWithoutBody(memberTokenDTO))
             .stream()
-            .map(e -> productRepository.findById(e.getProductId()))
-            .filter(Optional::isPresent)
-            .map(Optional::get)
+            .map(Wishlist::getProduct)
             .toList();
     }
 
     public void addWishlist(MemberTokenDTO memberTokenDTO, long productId) {
-        if (wishlistRepository.existsByMemberEmailAndProductId(memberTokenDTO.getEmail(),
-            productId)) {
-            throw new IllegalArgumentException("Wishlist already exists");
+        if (!productRepository.existsById(productId)) {
+            throw new IllegalArgumentException(PRODUCT_NOT_FOUND);
         }
```

Comment:
product가 존재하는지 검증 한 후에 wishlist를 추가해주고 있군요 💯 
DB를 조회해 존재여부를 검증할거라면 이렇게 사용해보는건 어떨까요?
아래 `isWishlistExists`에서도 적절히 사용할 수 있을 것 같고
질문주신 `from~~~DTOWithoutBody` 이런 변환 로직도 많이 줄일 수 있을 것 같습니다.
wishlist뿐만 아니라 다른 패키지의 repository, domain, service 모두 확인해보시고 개선하시면 좋겠습니다.

```suggestion
        Product product = productRepository.findById(productId).orElseThrow(() -> new IllegalArgumentException("적절한 예외사용"));
        Member member = memberRepository.findById(memberTokenDTO.getEmail()).orElseThrow(() -> new IllegalArgumentException("적절한 예외사용"));

```

Reply:
Product 객체의 경우 멘토님께서 말씀해주신 방법이 좋아보입니다. Member객체의 경우, JWT를 통해 값이 들어오고 있는 상황이라 별도의 검증로직 없이, Member DB를 접근하지 않게 하고 싶어서 EntityManager의 getReference로 임시의 Proxy객체를 만들어서 사용했는데 이 방법은 어떤가요,,,??

```
public void addWishlist(MemberTokenDTO memberTokenDTO, long productId) {
        Product product = productRepository.findById(productId)
            .orElseThrow(() -> new IllegalArgumentException(PRODUCT_NOT_FOUND));

        wishlistRepository.findByMemberEmailAndProductId(memberTokenDTO.getEmail(), productId)
            .ifPresent(
                e -> {
                    throw new IllegalArgumentException(WISHLIST_ALREADY_EXISTS);
                }
            );

        wishlistRepository.save(
            new Wishlist(
                product,
                entityManager.getReference(Member.class, memberTokenDTO.getEmail())
            )
        );
    }
```